integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,158 @@
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        offer = Offer.objects.filter(id=kwargs.get("pk")).first()
+        if offer is None and kwargs.get("pk") is not None:
+            raise Offer.DoesNotExist
+
+        (offer_version, offer_contact, offer_location, offer_organization) = (
+            (
+                offer.versions.first(),
+                offer.offer_contact,
+                offer.location,
+                offer.organization,
+            )
+            if offer is not None
+            else (None, None, None, None)
+        )
+        offer_form = OfferForm(instance=offer)
+        offer_version_form = OfferVersionForm(instance=offer_version)
+        contact_form = ContactForm(instance=offer_contact)
+        location_form = LocationForm(instance=offer_location)
+        organization_form = OrganizationForm(instance=offer_organization)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "offer_form": offer_form,
+                "offer_version_form": offer_version_form,
+                "contact_form": contact_form,
+                "location_form": location_form,
+                "organization_form": organization_form,
+            },
+        )
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Submit :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offer_form = OfferForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+        offer_version_form = OfferVersionForm(data=request.POST)
+        contact_form = ContactForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+        location_form = LocationForm(data=request.POST)
+        organization_form = OrganizationForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
```

Comment:
```suggestion
        if offer_id := kwargs.get("pk"):
            offer = Offer.objects.get(id=offer_id)
            offer_version = offer.versions.first()
            offer_contact = offer.offer_contact
            offer_location = offer.location
            offer_organization = offer.organization
        else:
            offer = offer_version = offer_contact = offer_location = offer_organization = None  # fmt: skip

        form_kwargs = {
            "data": request.POST,
            "additional_instance_attributes": {"creator": request.user},
        }

        offer_form = OfferForm(instance=offer, **form_kwargs)
        offer_version_form = OfferVersionForm(instance=offer_version, **form_kwargs)
        contact_form = ContactForm(instance=offer_contact, **form_kwargs)
        location_form = LocationForm(instance=offer_location, **form_kwargs)
        organization_form = OrganizationForm(instance=offer_organization, **form_kwargs)
```
You need to pass the `instance` for POST requests as well, otherwise you will just create new instances instead of editing existing ones. At this point, we can again think about de-duplicating the initialization process :laughing: